README.md

```diff
@@ -1,22 +1,47 @@
 # user-validator
 
+User-validator can be used to validate user data stored inside app-interface. 
 
 ## usage
 
 `user-validator validate --config config.yaml` 
 
-## Configuration
+You can either specify a configuration via _--config_ or set configuration via Environment variables.
 
-See the following example for a configuration.yaml. Here authtype is set to `token`. However, a token is not set and might be configurated via `VAULT_TOKEN` environment variable.
+### Yaml configuration
 
 ```YAML
 qontract: 
-  server_url: "http://localhost:4000/graphql"
+  serverurl: URL to the GraphQL API. REQUIRED
 
 vault:
-  addr: "http://localhost:8200"
-  authtype: "token"
+  addr: Address to access Vault. REQUIRED
+  authtype: Authentication type either token or approle REQUIRED
+  token: Token to access Vault, requires setting authtype to token
+  roleid: Role ID to use for authentication, requires setting authtype to approle 
+  secretid: Secret ID to use for authentication, requires setting authtype to approle 
 
-user_validation:
-  concurrent: 10
-```
\ No newline at end of file
+user_validator:
+  concurrency: Number of coroutines to use to query Github (default: Number of CPU cores)
```

Comment:
i'm not a fan of defaulting to the cores of the machine. can you default to the number of cores you have been granted by container `resources`, so essentially the resources of the cgroup?

Reply:
I chose that, cause it's an arbitrary number with some relation to the spec of the environment it is running in. An alternative is to just pick a number, like in the original code (10). 
But thinking about it now, setting it to a fixed number might be the better approach since the real bottleneck for this is the API request limit imposed by Github and not the CPU. Let me set it back to 10, as found in the original implementation.